waspc/src/Wasp/Generator/DbGenerator.hs

```diff
@@ -99,7 +110,49 @@ genPrismaSchema wasp = createTemplateFileDraft dstPath tmplSrcPath (Just templat
 
 genMigrationsDir :: Wasp -> Maybe FileDraft
 genMigrationsDir wasp =
-  (getMigrationsDir wasp) >>= \waspMigrationsDir ->
+  getMigrationsDir wasp >>= \waspMigrationsDir ->
     Just $ createCopyDirFileDraft (SP.castDir genProjectMigrationsDir) (SP.castDir waspMigrationsDir)
   where
     genProjectMigrationsDir = dbRootDirInProjectRootDir </> dbMigrationsDirInDbRootDir
+
+writeDbSchemaChecksumToFile :: Path' Abs (Dir ProjectRootDir) -> IO ()
+writeDbSchemaChecksumToFile genProjectRootDir = do
+  dbSchemaExists <- doesFileExist dbSchemaFp
+  when dbSchemaExists $ do
+    checksum <- calculateChecksumStringForFile dbSchemaFp
+    writeFile dbSchemaChecksumFp checksum
+  where
+    dbSchemaFp = SP.fromAbsFile $ genProjectRootDir </> dbSchemaFileInProjectRootDir
+    dbSchemaChecksumFp = SP.fromAbsFile $ genProjectRootDir </> dbSchemaChecksumFileInProjectRootDir
+
+checkIfDbSchemaMatchesChecksum :: Path' Abs (Dir ProjectRootDir) -> ExceptT String IO ()
+checkIfDbSchemaMatchesChecksum genProjectRootDir = do
+  dbSchemaExists <- liftIO $ doesFileExist dbSchemaFp
+  let maybeDbSchemaFp = if dbSchemaExists then Just dbSchemaFp else Nothing
+  dbSchemaChecksumExists <- liftIO $ doesFileExist dbSchemaChecksumFp
+  let maybeDbSchemaChecksumFp = if dbSchemaChecksumExists then Just dbSchemaChecksumFp else Nothing
```

Comment:
Yeah good call. I had it that way and went back and forth between combining with prep/use vs combining with names. Since you noticed too I'll consider a vote in that direction and swap back.